akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -1411,20 +1413,22 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
           updateLatestGossip(newGossip)
 
           val (exiting, nonExiting) = newlyDetectedUnreachableMembers.partition(_.status == Exiting)
-          if (nonExiting.nonEmpty)
+          nonExiting.foreach { node =>
             logWarning(
-              "Marking node(s) as UNREACHABLE [{}]. Node roles [{}]",
-              nonExiting.mkString(", "),
-              selfRoles.mkString(", "))
+              LogMarker("cluster.unreachable", Map(LogMarker.Properties.RemoteAddress -> node.address)),
+              "Marking node as UNREACHABLE [{}].",
+              node)
+          }
           if (exiting.nonEmpty)
             logInfo(
               "Marking exiting node(s) as UNREACHABLE [{}]. This is expected and they will be removed.",
               exiting.mkString(", "))
-          if (newlyDetectedReachableMembers.nonEmpty)
+          nonExiting.foreach { node =>
             logInfo(
-              "Marking node(s) as REACHABLE [{}]. Node roles [{}]",
-              newlyDetectedReachableMembers.mkString(", "),
-              selfRoles.mkString(","))
+              LogMarker("cluster.reachable", Map(LogMarker.Properties.RemoteAddress -> node.address)),
+              "Marking node as REACHABLE [{}].",
+              node)
+          }
```

Comment:
example of logging with `LogMarker`

Reply:
perhaps for the final version these could be public constants? To make them easy to discover / include in docs and see what other properties will be set

Reply:
You mean constants for the marker names?

We could define them like:
```scala
object ClusterLogMarker {
  /**
   * Some description and include the marker name and property names...
   */
  def reachable(remoteAddress: Address): LogMarker =
    LogMarker("cluster.reachable", Map(LogMarker.Properties.RemoteAddress -> node.address))
}
```

Then we could link to apidocs.

My thinking about the property names is that it's useful to reuse the same property name for several event types (markers) and not have too many properties that are unique for s single event type.

Because they all show up as a flat structure in Kibana. Could be useful to search or correlate for specific property name across different event types (markers), for example the `remoteAddress`.

Do we need to prefix the property names with `akka`? As we do with most other mdc values.

Reply:
I have moved the definitions of the log markers to `ClusterLogMarker` and `RemoteLogMarker`.